### ISSUE TITLE
Improve Mumble for blind person

### ISSUE THREAD

**Issue by citril** [mumble-voip/mumble#4069]
I am blind and I try to use Mumble but it is not planned for us. keyboard shortcuts should be added to access the menus

**Comment by Krzmbrzl** [comment]
I think we do have keyboard-navigation built into Mumble somehow, but I'm not sure we have documentation on that lying around...

@davidebeatrici do you know of some documentation on this?

**Comment by citril** [comment]
I did not find any documentation on this subject

**Comment by davidebeatrici** [comment]
@Krzmbrzl Unfortunately not.

The keyboard mappings are Qt's ones, but they can be optimized for better usability.

**Comment by Krzmbrzl** [comment]
Okay so we do have keyboard navigation but there's no doc on that? Now that's great xD

> but they can be optimized for better usability.

where?

**Comment by davidebeatrici** [comment]
See #3417 for an example.

**Comment by Krzmbrzl** [comment]
I put this in the `documentation` category as to my understanding the requested feature is already implemented, but undocumented.

Since I don't have a clue about this though, I'm afraid I can't really do this myself...

**Comment by Kissaki** [comment]
I am a little bit confused about what we are talking about here. What is not working or missing?

The standard menus do have keyboard access keys. For seeing people they can see them by pressing and holding ALT, and can use/activate them with ALT.

Tabbing through form items in the configure dialog also works.

These do not require documentation specifically. It is the general concept of keyboard navigability. We could document some general introduction and overview of these things and mechanics of course, but I do not see how OP was asking for that. If they are already a blind software user they will already know these standard mechanisms.

We definitely want to support accessibility including blind users. In the past there were some efforts or discussion, and also some concerns raised. And I do think some changes were made to improve use with screen readers back then (namely naming controls adequately). At one point we were at least hoping a Qt upgrade would help with accessibility improvements. To a degree we depend on that framework.

I would definitely like to see some concrete, actionable concerns and tasks. A general “does not work” is not helpful especially when I can definitely see *something* working.

An actual blind user as a test person and accessibility advocate/guide would definitely be helpful in doing so.

**Comment by Kissaki** [comment]
@citril I can definitely see the standard menus having accessibility keys and keyboard navigation working. What issues are you facing?

**Comment by citril** [comment]
Hello

I agree with your saying I will try to be clearer already in screen reader I use NVDA which is free of rights under windows, I invite you to install it and to listen to the result you must make as if you do not see the screen but only nvda guides you through the menus. Then some keyboard shortcuts are duplicated for example to access the audio configuration wizard you have to do ALT O + A if you do ALT O then alt A you will come across the help it is not really common. navigation with the tab key is no problem.
Then in the software there is still a lot of cursor with the mention potentiometer 9152 for example for the potentiometer of activation of the voice.

Thank you for the work already done and I am sorry to disturb you with minor technical problems.

Sincerely

Cyril Grither

 

De : Jan Klass <notifications@github.com> 
Envoyé : samedi 18 avril 2020 09:44
À : mumble-voip/mumble <mumble@noreply.github.com>
Cc : citril <cyril.grither@gmail.com>; Author <author@noreply.github.com>
Objet : Re: [mumble-voip/mumble] adapted mumble for blind personn (#4069)

 

I am a little bit confused about what we are talking about here. What is not working or missing?

The standard menus do have keyboard access keys. For seeing people they can see them by pressing and holding ALT, and can use/activate them with ALT.

Tabbing through form items in the configure dialog also works.

These do not require documentation specifically. It is the general concept of keyboard navigability. We could document some general introduction and overview of these things and mechanics of course, but I do not see how OP was asking for that. If they are already a blind software user they will already know these standard mechanisms.

We definitely want to support accessibility including blind users. In the past there were some efforts or discussion, and also some concerns raised. And I do think some changes were made to improve use with screen readers back then (namely naming controls adequately). At one point we were at least hoping a Qt upgrade would help with accessibility improvements. To a degree we depend on that framework.

I would definitely like to see some concrete, actionable concerns and tasks. A general “does not work” is not helpful especially when I can definitely see something working.

An actual blind user as a test person and accessibility advocate/guide would definitely be helpful in doing so.

**Comment by alexarnaud** [comment]
@citril I'm reading this discussion with some interests because I'm visual-impaired too running the Orca screen reader under Linux. Many thanks for trying to raise accessibility hear.

@citril What is written below is to help you and was written with a good intention.

To be fair and to help the Mumble team, I think this issue should be closed or converted as a meta issue.

@citril You could read this article to learn more on how to report an issue: https://www.chiark.greenend.org.uk/~sgtatham/bugs.html

Even if I'm visual-impaired myself, I don't understand precisely what issues you face and what you want to figure out. One bug report should contain one problem explained in details.

Hope this could help. It's really important to have blind people reporting issues, keep going.

**Comment by Krzmbrzl** [comment]
> To be fair and to help the Mumble team, I think this issue should be closed or converted as a meta issue.

Sounds good to me. I'll close this issue until it is proven that it should be re-opened :)
